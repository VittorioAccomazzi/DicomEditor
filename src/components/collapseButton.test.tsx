import { cleanup, render } from '@testing-library/react'
import React from 'react'
import CollapseButton from './collapseButton'

beforeEach(() => {
    // setup a DOM element as a render target
  });
  
  afterEach(()=>{
    cleanup()
    jest.clearAllMocks()
  })

test('shall default on close',()=>{
    const onExpand = ()=>{}
    const wrapper=render(<CollapseButton onExpand={onExpand}/>)
    const el = wrapper.getByRole('button') as HTMLButtonElement
    expect(el).toBeInTheDocument()
    expect(wrapper).toMatchSnapshot()
})

test('shall invoke callback',()=>{
    const onExpand = jest.fn()
    const wrapper=render(<CollapseButton onExpand={onExpand}/>)
    const el = wrapper.getByRole('button') as HTMLButtonElement
    el.click()
    expect(onExpand).toBeCalled()
    expect(onExpand).toBeCalledWith(true)
})