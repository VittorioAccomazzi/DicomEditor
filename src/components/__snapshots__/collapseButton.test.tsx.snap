// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shall default on close 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-expanded="false"
        aria-label="show more"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-expand-1 undefined"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      aria-expanded="false"
      aria-label="show more"
      class="MuiButtonBase-root MuiIconButton-root makeStyles-expand-1 undefined"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
